earth_enterprise/src/common/khxml/khxml.cpp

```diff
@@ -161,18 +227,38 @@ class UsingXMLGuard
 };
 
 static khMutexBase xmlLibLock = KH_MUTEX_BASE_INITIALIZER;
```

Comment:
Minor: for simplicity you might consider using a single mutex instead of the three you have here. Splitting them up like this probably performs better, but I'm not sure the difference is enough to outweigh the simplification of only having one.